portfolio/src/main/webapp/script.js

```diff
@@ -6,15 +6,65 @@ function changeImage() {
 }
 
 /**
-  * Creates and returns an HTML list element containing
-  * a string provided by the the text parameter.
-  * @param {String} text Content of the list element.
-  * @return {<li>} HTML list element with the contents of the text parameter.
+ * ENUM type to specify direction of vote.
+ */
+const Direction = {
+  up: 1,
+  down: -1
+}
+
+/**
+  * Creates and adds a post card containing a caption and 
+  * a score as provided by the post parameter.
+  * @param {!{comment: string, username: string, score: number}} 
   */
-function createListElement(text) {
-  const listElement = document.createElement('li');
-  listElement.innerText = text;
-  return listElement;
+function createListElement(post) {
+  let card = document.createElement("div");
+  card.className = "card m-1";
+
+  let body = document.createElement("div");
+  body.className = "card-body";
+
+  let comment = document.createElement("p");
+  comment.className = "card-text";
+  comment.innerHTML = post.comment;
+
+  let buttons = document.createElement("div");
+  buttons.className = "btn-group";
+  buttons.setAttribute("role", "group");
+
+  let minus = document.createElement("button");
+  minus.addEventListener("click", function () {
+    votePost(this, Direction.down);
+  });
+  minus.type = "button";
+  minus.className = "btn btn-sm btn-primary";
+  minus.innerHTML = " - ";
+
+  let score = document.createElement("button");
+  score.type = "button";
+  score.id = "score";
+  score.className = "btn btn-sm btn-outline-primary disabled";
+  score.innerHTML = post.score;
+
+  let plus = document.createElement("button");
+  plus.addEventListener("click", function () {
+    votePost(this, Direction.up);
+  });
+  plus.type = "button";
+  plus.className = "btn btn-sm btn-primary";
+  plus.innerHTML = " + ";
```

Comment:
I would factor this and the constructor of minus into a subroutine to save some code.